packages/dds/tree/src/shared-tree-core/commitEnricher.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { GraphCommit } from "../core/index.js";
+
+/**
+ * A component that can be used to enrich outbound commits.
+ */
+export interface CommitEnricher<TChange> {
+	/**
+	 * Enriches a commit with adequate refreshers.
+	 * Assumes that the commit will be submitted.
+	 * @param commit - the commit to enrich. Not mutated.
+	 * @param isResubmit - whether the commit is a resubmission of a previously submitted commit.
+	 * Must be true if and only if `isInResubmitPhase` is true.
+	 * @returns the enriched commit. Possibly the same as the one passed in.
+	 */
+	enrichCommit(commit: GraphCommit<TChange>, isResubmit: boolean): GraphCommit<TChange>;
```

Comment:
This was meant to be a safety net, simiarly to how some APIs like `open(): ScopeID` and `close(ScopeID): void` (when they allow nested scoping) force you to state which scope you think you're closing so it can yell at you early if your calling code is wrong.

I agree it's a little weird, but I'm really scared of `SharedObject` being out of sync with our internal state (e.g., they disagree about how many commits are going to be resubmitted), and if we don't get this error, we will likely get a document corruption bug that will be extremely hard to track down.

I'll remove this parameter and have SharedTreeCore assert that `isInResubmitPhase` always matches its expecations. I think that ticks all the boxes.